core/leader/leader.go

```diff
@@ -13,18 +13,29 @@ import (
 	"chain/log"
 )
 
-var (
-	isLeading bool
-	lock      sync.Mutex
-)
+type Leadership interface {
+	IsLeading() bool
+	Address(ctx context.Context) (string, error)
+}
 
-// IsLeading returns true if this process is
-// the core leader.
-func IsLeading() bool {
-	lock.Lock()
-	l := isLeading
-	lock.Unlock()
-	return l
+type leadership struct {
```

Comment:
Got it. An interface makes sense, but if possible it would be slightly better to define it in the place where it's consumed, and have this package still return concrete types.